src/artifact_service/service.rs

```diff
@@ -71,9 +73,54 @@ impl ArtifactService {
         package_type: PackageType,
         package_specific_id: String,
     ) -> Result<String, BuildError> {
-        self.build_event_client
-            .start_build(package_type, package_specific_id)
-            .await
+        debug!(
+            "Request build of {:?} {:?}",
+            package_type, package_specific_id
+        );
+
+        let local_peer_id = self.local_keypair.public().to_peer_id();
```

Comment:
The local PeerId can be retrieved from the p2p_client.

```suggestion
        let local_peer_id = self.p2p_client.local_peer_id;
```